Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Utilities/RouteTemplateUtility.cs

```diff
@@ -0,0 +1,114 @@
+    /// <param name="actualPath">The actual path to extract parameters from.</param>
+    /// <returns>A dictionary of extracted path parameters and their values.</returns>
+    /// <example>
+    /// Using this method:
+    /// <code>
+    /// var routeTemplate = "/users/{id}/orders/{orderId}";
+    /// var actualPath = "/users/123/orders/456";
+    /// var parameters = RouteTemplateUtility.ExtractPathParameters(routeTemplate, actualPath);
+    /// // parameters will contain: { {"id", "123"}, {"orderId", "456"} }
+    /// </code>
+    /// </example>
+    public static Dictionary<string, string> ExtractPathParameters(string routeTemplate, string actualPath)
+    {
+        // Preprocess the route template to convert from .net style format to aws
+        routeTemplate = PreprocessRouteTemplate(routeTemplate);
+
+        var template = TemplateParser.Parse(routeTemplate);
+        var matcher = new TemplateMatcher(template, new RouteValueDictionary());
+        var routeValues = new RouteValueDictionary();
+
+        if (matcher.TryMatch(actualPath, routeValues))
+        {
+            var result = new Dictionary<string, string>();
+
+            foreach (var param in template.Parameters)
+            {
+                if (routeValues.TryGetValue(param.Name, out var value))
+                {
+                    var stringValue = value?.ToString() ?? string.Empty;
+
+                    // For catch-all parameters, remove the leading slash if present
+                    if (param.IsCatchAll)
+                    {
+                        stringValue = stringValue.TrimStart('/');
+                    }
+
+                    // Restore original parameter name
+                    var originalParamName = RestoreOriginalParamName(param.Name);
+                    result[originalParamName] = stringValue;
+                }
+            }
+
+            return result;
+        }
+
+        return new Dictionary<string, string>();
+    }
+
+    /// <summary>
+    /// Preprocesses a route template to make it compatible with ASP.NET Core's TemplateMatcher.
+    /// </summary>
+    /// <param name="template">The original route template, potentially in AWS API Gateway format.</param>
+    /// <returns>A preprocessed route template compatible with ASP.NET Core's TemplateMatcher.</returns>
+    /// <remarks>
+    /// This method performs two main transformations:
+    /// 1. Converts AWS-style {proxy+} to ASP.NET Core style {*proxy}
+    /// 2. Handles AWS ignoring constraignts by temporarily renaming parameters
+    ///    (e.g., {abc:int} becomes {__aws_param__abc__int})
+    /// </remarks>
+    private static string PreprocessRouteTemplate(string template)
+    {
+        // Convert AWS-style {proxy+} to ASP.NET Core style {*proxy}
+        template = Regex.Replace(template, @"\{(\w+)\+\}", "{*$1}");
+
+        // Handle AWS-style "constraints" by replacing them with temporary parameter names
```

Comment:
My understanding is API Gateway does not support constraints like `int` in `{abs:int}`. So if a user gave us a resource path that including constraints when configuring the lambda functions for the emulator it seems like we should throw an error for bad user configuration.

Reply:
A heads up, as someone that has RegEx dyslexia I generally push back on having anything but the simplest of regexes. I do appreciate these were well commented.

Reply:
good question - i actually tried creating a path in api gateway called `{variable:int}` and this was a valid path, with the variable be named the literal `variable:int`. so this code basically makes it so that `variable:int` is correctly parsed from the path as the path parameter key. if we directly used the microsoft template parser, without any modifications, it would think the path parameter key is just `variable` that has an `int` constraint, which is not correct.

Reply:
i can update the comment to make it more clear if that helps - aws does not support using constraints, but they let you name the path parameter with same naming convention

Reply:
Interesting API Gateway allows you to have a variable with a semicolon.